### PR TITLE
Shellcraft functions will now have a single trailing newline

### PR DESCRIPTION
**Pull request by IdolfHatler**

### CODE REVIEW

File: pwnlib/shellcraft/internal.py [hunk 1]
```diff
@@ -88,10 +95,12 @@ def make_function(funcname, filename, directory):
     # but what would not have the right signature.
     # While we are at it, we insert the docstring too
     exec '''
-def wrap(template):
+def wrap(template, render_global):
     def %s(%s):
         %r
+        is_inside, render_global.is_inside = render_global.is_inside, True
```
Comment (zachriggle) [inline pwnlib/shellcraft/internal.py]: Seems like a good use for a context manager, no?

Comment (IdolfHatler) [inline pwnlib/shellcraft/internal.py]: It is not actually easier to write. But sure, I could rewrite it if you prefer. :)

**Comment by zachriggle** [comment]
No problem, I can do it later.  I was just concerned about throwing an exception during rendering and then borking it afterward

**Comment by IdolfHatler** [comment]
Oh, I hadn't considered exceptions. Derp.

I still think a finally block would be easier, though probably not prettier.

**Pull request merged**